### PR TITLE
verify #40

### PR DESCRIPTION
**Pull request by HagarMeir**
Signed-off-by: Hagar Meir <hagar.meir@ibm.com>

### CODE REVIEW

**Review by adecaro — CHANGES_REQUESTED** [review]

File: token/core/fabtoken/validator.go [hunk 1]
```diff
@@ -167,6 +170,12 @@ func (v *Validator) verifyIssue(issue driver.IssueAction) error {
 }
 
 func (v *Validator) VerifyTransfer(inputTokens []*token2.Token, tr driver.TransferAction) error {
+	if inputTokens[0] == nil {
```
Comment (KElkhiyaoui) [inline token/core/fabtoken/validator.go:173]: Before checking if the inputTokens[0] is not nil, we should check that len(inputTokens) is not 0.

File: token/core/fabtoken/validator.go [hunk 2]
```diff
@@ -152,10 +152,48 @@ func (v *Validator) VerifyTransfers(ledger driver.Ledger, transferActions []*Tra
...
+	}
 	return nil
 }
 
 func (v *Validator) VerifyTransfer(inputTokens []*token2.Token, tr driver.TransferAction) error {
+	typ := inputTokens[0].Type
+	inputSum := token2.NewZeroQuantity(64)
+	outputSum := token2.NewZeroQuantity(64)
+	for _, input := range inputTokens {
+		q, err := token2.ToQuantity(input.Quantity, 64)
```
Comment (KElkhiyaoui) [inline token/core/fabtoken/validator.go:183]: We should check that each input is not nil to avoid panic.

File: token/core/fabtoken/validator.go [hunk 3]
```diff
@@ -170,16 +170,22 @@ func (v *Validator) verifyIssue(issue driver.IssueAction) error {
 }
 
 func (v *Validator) VerifyTransfer(inputTokens []*token2.Token, tr driver.TransferAction) error {
-	if inputTokens[0] == nil {
-		errors.Errorf("there is no input")
-	}
 	if tr.NumOutputs() == 0 {
 		errors.Errorf("there is no output")
```
Comment (KElkhiyaoui) [inline token/core/fabtoken/validator.go:174]: return is missing

File: token/core/fabtoken/validator.go [hunk 4]
```diff
@@ -170,16 +170,22 @@ func (v *Validator) verifyIssue(issue driver.IssueAction) error {
...
 	}
+	if len(inputTokens) == 0 {
+		errors.Errorf("there is no input")
```
Comment (KElkhiyaoui) [inline token/core/fabtoken/validator.go:177]: return is missing

File: token/core/fabtoken/validator.go [hunk 5]
```diff
@@ -170,16 +170,22 @@ func (v *Validator) verifyIssue(issue driver.IssueAction) error {
...
+	}
+	if inputTokens[0] == nil {
+		errors.Errorf("first input is nil")
```
Comment (KElkhiyaoui) [inline token/core/fabtoken/validator.go:180]: return is missing

**Review by KElkhiyaoui — CHANGES_REQUESTED** [review]

**Review by KElkhiyaoui — APPROVED** [review]

**Review by KElkhiyaoui — APPROVED** [review]

**Review by KElkhiyaoui — APPROVED** [review]

**Review by adecaro — APPROVED** [review]

**Pull request merged**